Assets/MRTK/Core/Utilities/MixedRealityPlayspace.cs

```diff
@@ -27,7 +27,10 @@ public static void Destroy()
             // Playspace makes main camera dependent on it (see Transform initialization),
             // so here it needs to restore camera's initial position. 
             // Without second parameter camera will not move to its original position.
-            CameraCache.Main.transform.SetParent(null, false);
+            if (CameraCache.Main.transform.parent == mixedRealityPlayspace)
+            {
+                CameraCache.Main.transform.SetParent(null, false);
```

Comment:
I think we still need some mechanism for detaching everything that's currently under the playspace. Otherwise, the next line (`UnityEngine.Object.Destroy(mixedRealityPlayspace.gameObject);`) will destroy the playspace and all its children, including any custom hierarchies. From reading your description, I'm assuming the camera is still under the playspace _somewhere_, even if it isn't a first tier child, which means it'll be destroyed too.

Reply:
Yeah the camera definitely needs to stay under the playpsace. 

Here is how it currently looks
![image](https://user-images.githubusercontent.com/7420990/83447223-1a926f80-a41e-11ea-8985-7e84424b2ad3.png)

Reply:
One thing to consider with this, is that I'd like to instantiate the Oculus rig at runtime so that all these oculus specific scripts aren't included in the scene if you're not working with Oculus. 
Ideally Id be able to destroy the camera currently in place, and replace it with it with this rig at runtime, but it seems the gaze pointer leaves behind dangling references when destroyed at runtime.

Reply:
Regarding your point on destroying whats under the playspace, I don't think that's really a problem in my case, but I do agree that considering how to properly detach things makes sense. I just didn't wanna do some aribtrary detaching that could in fact hinder the destruction process.

Reply:
@provencher, have you taken a look at how the UnityAR camera settings provider manages required camera and playspace components? 

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/releases/2.4.0/Assets/MRTK/Providers/UnityAR/UnityARCameraSettings.cs

Note, the above link is to the 2.4.0 implementation